### DISCUSSION TITLE
Lambda and delegate types

### DISCUSSION THREAD

**Discussion by GeroL** [dadhi/FastExpressionCompiler#318] [Q&A]
Hi,

how do I write lambda function as Expressions?
I have this code: 

`Expression<Func<double, bool>> exprCast = (param) => System.Convert.ToBoolean(param);`

It gives me this error:
Cannot convert anonymous method block to type 'type' because it is not a delegate type

**Comment by dadhi** [comment]
I don't see the problem https://dotnetfiddle.net/eSKXs0

**Comment by GeroL** [comment]
Yes it works with System.Linq.Expressions but not with FEC

**Comment by dadhi** [comment]
FEC is orthogonal to the hoisted lambda, because it is a C# compiler feature. Could you provide the full test to see what's failing?
